bpf/lib/nat.h

```diff
@@ -1203,13 +1237,13 @@ snat_v4_rev_nat(struct __ctx_buff *ctx, const struct ipv4_nat_target *target, __
 
 	if (snat_v4_rev_nat_can_skip(target, &tuple))
 		return NAT_PUNT_TO_STACK;
-	ret = snat_v4_rev_nat_handle_mapping(ctx, &tuple, &state, off, target, ext_err);
+	ret = snat_v4_rev_nat_handle_mapping(ctx, &tuple, has_l4_header, ct_action, &state,
+					     off, target, ext_err);
 	if (ret > 0)
 		return CTX_ACT_OK;
 	if (ret < 0)
 		return ret;
 
-rewrite_ingress:
 	return snat_v4_rewrite_ingress(ctx, &tuple, state, off);
```

Comment:
Note that this is currently missing a `has_l4_header` parameter. So I think it would attempt to revSNAT the L4 port even for a fragment.